### ISSUE TITLE
[BUG]Reservation can not be used with pvc in ReadWriteOncePod mode

### ISSUE THREAD

**Issue by KunWuLuan** [koordinator-sh/koordinator#923]
Volumerestriction will check volumerefcount in nodeinfo in prefilter.
So real pod will have conflicts with reserve pod. 

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- App version:
- Kubernetes version (use `kubectl version`):
- Install details (e.g. helm install args):
- Node environment (for koordlet/runtime-proxy issue):
  - Containerd/Docker version:
  - OS version:
  - Kernal version:
  - Cgroup driver: cgroupfs/systemd
- Others:

**Comment by saintube** [comment]
@KunWuLuan Yes, the `NodeInfo.PVCRefCounts` is missing processed in the Reservation Hooks. As we discussed, we may use `nodeInfo.AddPodInfo` and `framework.RunPreFilterExtensionAddPod` to get compatible with the native plugins.
cc @eahydra

**Comment by KunWuLuan** [comment]
> @KunWuLuan Yes, the `NodeInfo.PVCRefCounts` is missing processed in the Reservation Hooks. As we discussed, we may use `nodeInfo.AddPodInfo` and `framework.RunPreFilterExtensionAddPod` to get compatible to the native plugins. cc @eahydra

we have to modify the nodeinfo in prefilter hook. modification in prefilter won't work because volume restriction judge whether a pod can be scheduled in prefilter and volume restriction will be processed before custom plugins if we don't pass  a custom configuration

**Comment by eahydra** [comment]
> @KunWuLuan Yes, the `NodeInfo.PVCRefCounts` is missing processed in the Reservation Hooks. As we discussed, we may use `nodeInfo.AddPodInfo` and `framework.RunPreFilterExtensionAddPod` to get compatible to the native plugins. cc @eahydra

In this scenario, because the `Volumerestriction` plugin is checked in the PreFilter stage, we need to process the data in NodeInfo before the `Volumerestriction.PreFilter` is executed, so the existing Reservation Hook should be retained.

**Comment by saintube** [comment]
@KunWuLuan @eahydra We can process the nodeInfo either in the PreFilterHook or in the PreFilter stage with plugin orders configured. I think we can resolve this issue by replacing the current implementation in `preparePreFilterNodeInfo()` with the `nodeInfo.AddPodInfo` and `framework.RunPreFilterExtensionAddPod`.

**Comment by KunWuLuan** [comment]
I think it is OK to resolve this issue by replacing the current implementation in preparePreFilterNodeInfo() with the nodeInfo.AddPodInfo and framework.RunPreFilterExtensionAddPod. In that case, we have to remove filter hook from current implementation. If you think this way is OK, I will try to finish the work this week. @saintube @eahydra

**Comment by saintube** [comment]
@KunWuLuan I agree with it. WDYT @eahydra

**Comment by saintube** [comment]
> I think it is OK to resolve this issue by replacing the current implementation in preparePreFilterNodeInfo() with the nodeInfo.AddPodInfo and framework.RunPreFilterExtensionAddPod. In that case, we have to remove filter hook from current implementation. If you think this way is OK, I will try to finish the work this week. @saintube @eahydra

@KunWuLuan I agree with it. WDYT @eahydra

**Comment by eahydra** [comment]
If the Reservation Hook is completely removed, similar capabilities can be achieved by adjusting the plugin configuration order of the PreFilter, which means that the Reservation must be configured before the VolumeRestriction plugin, which will bring configuration complexity to the user. If the complexity of this configuration can be reduced, I think it is OK.
